schemas/descriptors/label/descriptorLabel.schema.json

```diff
@@ -0,0 +1,77 @@
+{
+  "meta:license": [
+    "Copyright 2021 Adobe Systems Incorporated. All rights reserved.",
+    "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/common/descriptors/descriptorLabel",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Label Descriptor",
+  "meta:extensible": false,
+  "meta:extends": [
+    "https://ns.adobe.com/xdm/common/desciptors/schemadescriptor"
+  ],
+  "meta:abstract": false,
+  "type": "object",
+  "description": "Describes a label at the field level for a given class/fieldgroup/schema",
+  "definitions": {
+    "descriptorLabel": {
+      "properties": {
+        "@type": {
+          "type": "string",
+          "const": "xdm:descriptorLabel",
+          "meta:xdmType": "string"
+        },
+        "xdm:labels": {
+          "title": "Labels",
+          "description": "When present, it allows an array of objects with three properties such as labelType, labelName and labelValue",
+          "type": "array",
+          "items": {
+            "type": "object",
+            "meta:xdmType": "object",
+            "required": [
+              "labelType",
+              "labelName",
+              "labelValue"
+            ],
+            "properties": {
+              "labelType": {
+                "type": "string"
+              },
+              "labelName": {
+                "type": "string"
+              },
+              "labelValue": {
+                "type": "string"
+              }
+            }
+          },
+          "meta:xdmType": "array"
+        }
+      },
+      "oneOf" : [
```

Comment:
What is the purpose of the "oneOf" here? Are we saying this descriptor can have have either a xdm:labels OR a xdm:sourceProperty? 

Seems like you want both of those to always be required so simply do:
"required": ["xdm:sourceProperty", "xdm:labels"]

Reply:
Yes, Taken care of. I re-used the relationship descriptor schema definitions that defines this way. I should given a closer look.